File: src/main/java/com/example/demo/service/EventService.java
package com.example.demo.service;

import com.example.demo.model.Event;
import com.example.demo.model.User;
import com.example.demo.repository.EventRepository;
import com.example.demo.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
@Service
public class EventService {
    
    @Autowired
    private EventRepository eventRepository;
    
    @Autowired
    private UserRepository userRepository;

    public Event createEvent(Event event) {
        // organizer exists?
        Optional<User> organizer = userRepository.findById(event.getOrganizerId());
        if (organizer.isEmpty()) {
            throw new RuntimeException("Organizer not found with ID: " + event.getOrganizerId());
        }
        
        // organizer name 
        User user = organizer.get();
        event.setOrganizerName(user.getFirstName() + " " + user.getLastName());
        
        return eventRepository.save(event);
    }
    
    //all
    public List<Event> getAllEvents() {
        return eventRepository.findAll();
    }
    
    // ID
    public Optional<Event> getEventById(Long id) {
        return eventRepository.findById(id);
    }
    
    //organizer
    public List<Event> getEventsByOrganizer(Long organizerId) {
        return eventRepository.findByOrganizerId(organizerId);
    }
    
    // Update
    public Event updateEvent(Long id, Event eventDetails) {
        Optional<Event> existingEvent = eventRepository.findById(id);
        if (existingEvent.isEmpty()) {
            throw new RuntimeException("Event not found with ID: " + id);
        }
        
        Event event = existingEvent.get();
        event.setTitle(eventDetails.getTitle());
        event.setDescription(eventDetails.getDescription());
        event.setEventDate(eventDetails.getEventDate());
        event.setLocation(eventDetails.getLocation());
        event.setCapacity(eventDetails.getCapacity());
        
        return eventRepository.save(event);
    }
    
    // Delete 
    public void deleteEvent(Long id) {
        if (!eventRepository.existsById(id)) {
            throw new RuntimeException("Event not found with ID: " + id);
        }
        eventRepository.deleteById(id);
    }
    
    // S title
    public List<Event> searchEventsByTitle(String title) {
        return eventRepository.findByTitleContainingIgnoreCase(title);
    }
    
    // S  location
    public List<Event> searchEventsByLocation(String location) {
        return eventRepository.findByLocationContainingIgnoreCase(location);
    }
}

File: user-frontend/src/services/eventService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/events';

const eventService = {
  // Get all events
  getAllEvents: async () => {
    try {
      const response = await axios.get(API_BASE_URL, {
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        }
      });
      return response.data;
    } catch (error) {
      console.error('Error fetching events:', error);
      throw new Error('Failed to fetch events');
    }
  },

  // Get event by ID
  getEventById: async (id) => {
    try {
      const response = await axios.get(`${API_BASE_URL}/${id}`);
      return response.data;
    } catch (error) {
      throw new Error('Failed to fetch event');
    }
  },

  // Get events by organizer
  getEventsByOrganizer: async (organizerId) => {
    try {
      const response = await axios.get(`${API_BASE_URL}/organizer/${organizerId}`);
      return response.data;
    } catch (error) {
      throw new Error('Failed to fetch events by organizer');
    }
  },

  // Create new event
  createEvent: async (event) => {
    try {
      const response = await axios.post(API_BASE_URL, event, {
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        }
      });
      return response.data;
    } catch (error) {
      // Extract error message from backend response
      const errorMessage = error.response?.data || error.message || 'Failed to create event';
      throw new Error(errorMessage);
    }
  },

  // Update event
  updateEvent: async (id, event) => {
    try {
      const response = await axios.put(`${API_BASE_URL}/${id}`, event);
      return response.data;
    } catch (error) {
      throw new Error('Failed to update event');
    }
  },

  // Delete event
  deleteEvent: async (id) => {
    try {
      await axios.delete(`${API_BASE_URL}/${id}`);
      return true;
    } catch (error) {
      throw new Error('Failed to delete event');
    }
  },

  // Search events by title
  searchEventsByTitle: async (title) => {
    try {
      const response = await axios.get(`${API_BASE_URL}/search/title/${title}`);
      return response.data;
    } catch (error) {
      throw new Error('Failed to search events by title');
    }
  },

  // Search events by location
  searchEventsByLocation: async (location) => {
    try {
      const response = await axios.get(`${API_BASE_URL}/search/location/${location}`);
      return response.data;
    } catch (error) {
      throw new Error('Failed to search events by location');
    }
  }
};

export default eventService;
